Enhancement implementation of #141

emrekara37 opened the pull request:
Hi @justinyoo 

i implemented enhancement mentioned in this issue,  i hope you will like it 

Resolves #141

justinyoo commented on src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Configurations/DefaultOpenApiCustomUIOptions.cs:
```diff
@@ -29,10 +31,10 @@ public DefaultOpenApiCustomUIOptions(Assembly assembly)
-        /// <inheritdoc/>
-        public virtual async Task<string> GetStylesheetAsync()
+
+        private async Task<string> ReadFromStream(string path)
```
Can we move this private method below the public methods?

justinyoo commented on src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Configurations/DefaultOpenApiCustomUIOptions.cs:
```diff
@@ -46,21 +48,34 @@ public virtual async Task<string> GetStylesheetAsync()
             }
         }
 
+        private async Task<string> ReadFromUri(Uri uri)
```
Can we move this private method below the public methods?

justinyoo commented on src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Configurations/DefaultOpenApiCustomUIOptions.cs:
```diff
@@ -46,21 +48,34 @@ public virtual async Task<string> GetStylesheetAsync()
+        {
+            using (var client = new HttpClient())
+            {
+                var response = await client.GetAsync(uri);
+                response.EnsureSuccessStatusCode();
+                return await response.Content.ReadAsStringAsync();
```
This `EnsureSuccessStatusCode()` is supposed to handle exceptions. We don't need exception handling here. Instead, you can use `GetStringAsync(uri)` so that it returns empty (or null) string if the GET method returns 400+ HTTP status code.

justinyoo reviewed (changes requested):
Thanks @emrekara37 for the PR! I've left a few comments for you to review.

emrekara37 commented on src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Configurations/DefaultOpenApiCustomUIOptions.cs:
```diff
@@ -46,21 +48,34 @@ public virtual async Task<string> GetStylesheetAsync()
 
+        private async Task<string> ReadFromUri(Uri uri)
+        {
+            using (var client = new HttpClient())
```
i updated it using  static httpclient

emrekara37 commented:
> Thanks @emrekara37 for the PR! I've left a few comments for you to review.

i updated what you want

> Thanks @emrekara37 for the PR! I've left a few comments for you to review.

i updated

justinyoo reviewed (changes requested):
Thanks for the update. It LGTM, except...

Can you also please add test cases of your changes - `GetStylesheetAsync(...)` and `GetJavaScriptAsync(...)` to [this test class](https://github.com/Azure/azure-functions-openapi-extension/blob/main/test/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Tests/Configurations/DefaultOpenApiCustomUIOptionsTests.cs) by creating a `FakeUriCustomUIOptions` class that has references to external URL for CSS and JS?

emrekara37 commented:
> Thanks for the update. It LGTM, except...
> 
> Can you also please add test cases of your changes - `GetStylesheetAsync(...)` and `GetJavaScriptAsync(...)` to [this test class](https://github.com/Azure/azure-functions-openapi-extension/blob/main/test/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Tests/Configurations/DefaultOpenApiCustomUIOptionsTests.cs) by creating a `FakeUriCustomUIOptions` class that has references to external URL for CSS and JS?

Added test cases

justinyoo commented:
Thanks @emrekara37 for your contribution!
